cpp/src/gandiva/precompiled/arithmetic_ops.cc

```diff
@@ -469,6 +469,28 @@ BITWISE_NOT(int64)
 #undef DATE_TYPES
 #undef NUMERIC_BOOL_DATE_TYPES
 #undef NUMERIC_DATE_TYPES
+
+#define SIGN(TYPE)                         \
+  FORCE_INLINE                             \
+  gdv_##TYPE sign_##TYPE(gdv_##TYPE in1) { \
+    gdv_##TYPE out;                        \
+    if (in1 > 0) {                         \
+      out = static_cast<gdv_##TYPE>(1);    \
+    } else if (in1 < 0) {                  \
+      out = static_cast<gdv_##TYPE>(-1);   \
+    } else {                               \
+      out = static_cast<gdv_##TYPE>(0);    \
```

Comment:
instead of returning 0 return in1..this would take care of NaN. Also add unit tests for +infy, -infy, nan, 0.0f and -0.0f

Reply:
Ok